test/core/client_channel/lb_policy/xds_override_host_test.cc

```diff
@@ -166,6 +166,80 @@ TEST_F(XdsOverrideHostTest, SubchannelsComeAndGo) {
   ExpectRoundRobinPicks(picker.get(), {kAddresses[1]}, call_attributes);
 }
 
+TEST_F(XdsOverrideHostTest, FailedSubchannelIsNotPicked) {
+  // Send address list to LB policy.
+  const std::array<absl::string_view, 3> kAddresses = {
+      "ipv4:127.0.0.1:441", "ipv4:127.0.0.1:442", "ipv4:127.0.0.1:443"};
+  auto picker = ExpectStartupWithRoundRobin(kAddresses);
+  ASSERT_NE(picker, nullptr);
+  // Check that the host is overridden
+  std::map<UniqueTypeName, std::string> pick_arg{
+      {XdsHostOverrideTypeName(), "127.0.0.1:442"}};
+  EXPECT_EQ(ExpectPickComplete(picker.get(), pick_arg), kAddresses[1]);
+  auto subchannel = FindSubchannel(kAddresses[1]);
+  ASSERT_NE(subchannel, nullptr);
+  subchannel->SetConnectivityState(GRPC_CHANNEL_TRANSIENT_FAILURE,
+                                   absl::ResourceExhaustedError("Hmmmm"));
+  ExpectReresolutionRequest();
+  picker = ExpectState(GRPC_CHANNEL_READY);
+  ExpectRoundRobinPicks(picker.get(), {kAddresses[0], kAddresses[2]});
+}
+
+TEST_F(XdsOverrideHostTest, SubchannelConnectingIsQueued) {
+  // Send address list to LB policy.
+  const std::array<absl::string_view, 3> kAddresses = {
+      "ipv4:127.0.0.1:441", "ipv4:127.0.0.1:442", "ipv4:127.0.0.1:443"};
+  auto picker = ExpectStartupWithRoundRobin(kAddresses);
+  ASSERT_NE(picker, nullptr);
+  // Check that the host is overridden
+  std::map<UniqueTypeName, std::string> pick_arg{
+      {XdsHostOverrideTypeName(), "127.0.0.1:442"}};
+  EXPECT_EQ(ExpectPickComplete(picker.get(), pick_arg), kAddresses[1]);
+  auto subchannel = FindSubchannel(kAddresses[1]);
+  ASSERT_NE(subchannel, nullptr);
+  subchannel->SetConnectivityState(GRPC_CHANNEL_IDLE);
+  ExpectReresolutionRequest();
+  subchannel->SetConnectivityState(GRPC_CHANNEL_CONNECTING);
+  ExpectState(GRPC_CHANNEL_READY);
+  picker = ExpectState(GRPC_CHANNEL_READY);
+  ASSERT_NE(picker, nullptr);
+  EXPECT_EQ(ExpectPickComplete(picker.get()), kAddresses[0]);
+  ExpectPickQueued(picker.get(), pick_arg);
+}
+
+TEST_F(XdsOverrideHostTest, UsingRingHashAsChild) {
```

Comment:
Looking at this code has made me realize that we don't actually get any benefit from testing with ring_hash.  The main difference between round_robin and ring_hash is that ring_hash won't immediately trigger a reconnection request when the subchannel transitions from READY to IDLE, and we do want a test to verify that the xds_override_host policy will trigger the reconnection even if the child policy doesn't.  However, I don't think we need the ring_hash policy to test that case, because we will also need to do that in the case where the EDS health state is DRAINING, where the child policy does not even know about the subchannel in the first place.  So once we add support for EDS health states, we can test what we need even with round_robin.

I do think that we should ultimately have some tests with ring_hash as the child policy, because that is a case that we want to make sure will work, but I think we should do that in the e2e tests, not here in this unit test.  So I suggest removing this test.

Note that removing this test will also unblock this PR.  I still need to find a better approach to solve the `ExecCtx` problem here, but I probably won't get to that until after the holidays.  However, I would prefer not to merge this PR with the `ExecCtx` declared in the test code as it currently is (I suggested that as a quick hack to allow you to make progress, but I would prefer not to merge the code that way), so if we keep this test, that would delay this PR.